### PR TITLE
Add 0.5.5, 0.6.0, 0.6.1 C3 versions. Remove incorrect keywords from C3.

### PR DESCRIPTION
**Pull request by lerno**
This updates C3 with 0.5.5/0.6.0/0.6.1 and removes some long gone keywords. This also renames "0.4 -> 0.4.0" and "0.5 -> 0.5.0". The new versions are in the infra pull request here: https://github.com/compiler-explorer/infra/pull/1374

### CODE REVIEW

File: etc/config/c3.amazon.properties [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
 group.c3c.baseName=c3
 
-compiler.c3c04.semver=0.4
+compiler.c3c04.semver=0.4.0
 compiler.c3c04.exe=/opt/compiler-explorer/c3-0.4/c3c
 
-compiler.c3c05.semver=0.5
-compiler.c3c05.exe=/opt/compiler-explorer/c3-0.5/c3c
+compiler.c3c050.semver=0.5.0
+compiler.c3c050.exe=/opt/compiler-explorer/c3-0.5/c3c
```
Comment (partouf) [inline etc/config/c3.amazon.properties:15]: should add `compiler.c3c050.alias=c3c05`

Comment (lerno) [reply etc/config/c3.amazon.properties:15]: Updated

Comment (lerno) [reply etc/config/c3.amazon.properties:15]: Do you need anything else?

Comment (mattgodbolt) [reply etc/config/c3.amazon.properties:15]: This should be ok, thanks for the update.

**Review by mattgodbolt — APPROVED** [review]

**Pull request merged**